src/PointwiseFunctions/Xcts/LongitudinalOperator.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Elasticity/PotentialEnergy.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Xcts {
+
+/// \cond
+template <typename DataType>
+void longitudinal_operator(const gsl::not_null<tnsr::II<DataType, 3>*> result,
+                           const tnsr::ii<DataType, 3>& strain,
+                           const tnsr::II<DataType, 3>& inv_metric) noexcept {
+  std::fill(result->begin(), result->end(), 0.);
```

Comment:
In this particular case, owing two the double-loop inside, this *triples* the length of the code, with the actual formulae appearing three times.  Is this really the generally advised way, @moxcodes ?

Reply:
Hmm, I agree that this does make the code significantly more complicated, and looking around we haven't managed to be really consistent with this 'best-practice'.
One alternative that I see in a few places is to 
```
if(UNLIKELY(k == 0 and l == 0)) {
  /*do the calculation with an assign*/
} else {
  /*do the calculation with the +=*/
}
```
Which saves a bit on the unrolling, and for any vector type I'd imagine the `if` to be way cheaper than the copy initialization.

Reply:
If both of you are fine with the current code I suggest we just go with that. I don't mind the code duplication much in this case because it's an optimization in a function that probably won't change much and it's commented.